packages/worker/native/src/lib.rs

```diff
@@ -13,16 +16,20 @@ use temporal_sdk_core::{
 
 type BoxedWorker = JsBox<Arc<Worker>>;
 
+pub struct PollItem {
+    queue_name: String,
+    callback: Root<JsFunction>,
+}
+
 pub struct Worker {
     core: Box<dyn Core + Send + Sync>,
-    condition: Condvar,
-    suspended: Mutex<bool>,
+    sender: SyncSender<PollItem>,
 }
 
 impl Finalize for Worker {}
 
 impl Worker {
-    pub fn new() -> Self {
+    pub fn new(sender: SyncSender<PollItem>) -> Self {
```

Comment:
I find it a little easier to work with constructors like this where they create the channel inside of themselves, and return both `Self` and the receiving half. That way it can control it's own capacity etc